docs/tutorial/native-code-and-electron.md

```diff
@@ -0,0 +1,375 @@
+# Native Code and Electron
+
+One of Electron's most powerful features is the ability to combine web technologies with native code - both for compute-intensive logic as well as for the occasional native user interface, where desired.
+
+Electron does so by building on top of "Native Node.js Addons". You've probably already come across a few of them - packages like the famous [sqlite](https://www.npmjs.com/package/sqlite3) use native code to combine JavaScript and native technologies. You can use this feature to extend your Electron application with anything a fully native application can do:
+
+* Access native platform APIs not available in JavaScript. Any macOS, Windows, or Linux operating system API is available to you.
+* Create UI components that interact with native desktop frameworks.
+* Integrate with existing native libraries.
+* Implement performance-critical code that runs faster than JavaScript.
+
+Native Node.js addons are dynamically-linked shared objects (on Unix-like systems) or DLL files (on Windows) that can be loaded into Node.js or Electron using the `require()` or `import` functions. They behave just like regular JavaScript modules but provide an interface to code written in C++, Rust, or other languages that can compile to native code.
+
+# Tutorial: Creating a Native Node.js Addon for Electron
+
+This tutorial will walk you through building a basic Node.js native addon that can be used in Electron applications. We'll focus on concepts common to all platforms, using C++ as the implementation language. Once you complete this tutorial common to all native Node.js addons, you can move on to one of our platform-specific tutorials.
+
+This tutorial assumes you have Node.js and npm installed, as well as the basic tools necessary for compiling code on your platform (like Visual Studio on Windows, Xcode on macOS, or GCC/Clang on Linux). You can find detailed instructions in the [`node-gyp` readme](https://github.com/nodejs/node-gyp?tab=readme-ov-file).
+
+### Requirements: macOS
+
+To build native Node.js addons on macOS, you'll need the Xcode Command Line Tools. These provide the necessary compilers and build tools (namely, `clang`, `clang++`, and `make`). The following command will prompt you to install the Command Line Tools if they aren't already installed.
+
+```sh
+xcode-select --install
+```
+
+### Requirements: Windows
+
+The official Node.js installer offers the optional installation of "Tools for Native Modules", which installs everything required for the basic compilation of C++ modules - specifically, Python 3 and the "Visual Studio Desktop development with C++" workload. Alternatively, you can use `chocolatey`, `winget`, or the Windows Store.
```

Comment:
You do not need Visual Studio / Build Tools on Windows to compile correctly?

Reply:
Windows cannot compile C++ without an installed C++ compiler, so you do need those tools. Do you want to mention an alternative?

Reply:
Sorry I was not clear in my question, yes we need those tools to enable C++ compile, but I was suprised that Node.js installer offering thoses tools, I always downloaded manually VS2022 with C++ Compilation on my side 😄

Reply:
Ah, they do! They added that in light of https://www.npmjs.com/package/windows-build-tools, which I have since deprecated.